SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDevAuthTokenGenerator.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDevAuthTokenGenerator.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+
+#include "SpatialCommandUtils.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDevAuthTokenGenerator);
+
+FSpatialGDKDevAuthTokenGenerator::FSpatialGDKDevAuthTokenGenerator()
+	: bIsGenerating(false)
+{
+}
+
+void FSpatialGDKDevAuthTokenGenerator::DoUpdateSettings(FString DevAuthToken)
+{
+	AsyncTask(ENamedThreads::GameThread, [this, DevAuthToken]()
+		{
+			USpatialGDKEditorSettings* GDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+			GDKEditorSettings->DevelopmentAuthenticationToken = DevAuthToken;
+			GDKEditorSettings->SaveConfig();
+			GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
+
+			// Ensure we enable bUseDevelopmentAuthenticationFlow when using cloud deployment flow.
+			USpatialGDKSettings* GDKRuntimeSettings = GetMutableDefault<USpatialGDKSettings>();
+			GDKRuntimeSettings->bUseDevelopmentAuthenticationFlow = true;
+			GDKRuntimeSettings->DevelopmentAuthenticationToken = DevAuthToken;
```

Comment:
we are now doing a mix of assigning the variables directly and using the Set... methods available in the editor settings. Either remove line 26 or do
```
GDKEditorSettings->SetRuntimeDevelopmentAuthenticationToken();
GDKEditorSettings->SetRuntimeUseDevelopmentAuthenticationFlow()
```
and remove lines 28 - 31

I personally would prefer using the methods.